pyquil/api/compiler.py

```diff
@@ -161,3 +172,67 @@ def get_job_fn():
                             ping_time if ping_time else self.ping_time,
                             status_time if status_time else self.status_time)
+
+    def _clifford_application_payload(self, clifford, pauli):
+        """
+        Prepares a JSON payload for conjugating a Pauli by a Clifford - see apply_clifford_to_pauli.
+
+        :param Program clifford: A Program that consists only of Clifford operations.
+        :param PauliTerm pauli: A PauliTerm to be acted on by clifford via conjugation.
+        :return: The JSON payload, with keys "clifford" and "pauli".
+        """
+        pauli_string = pauli.pauli_string(0, max(clifford.get_qubits().union(pauli.get_qubits())))
+        payload = {"clifford": clifford.out(),
+                   "pauli": pauli_string}
+        return payload
+
+    def apply_clifford_to_pauli(self, clifford, pauli_in):
+        """
+        Given a circuit that consists only of elements of the Clifford group, return its action on a PauliTerm.
+
+        In particular, for Clifford C, and Pauli P, this returns the PauliTerm representing PCP^{\dagger}.
+
+        :param Program clifford: A Program that consists only of Clifford operations.
+        :param PauliTerm pauli_in: A PauliTerm to be acted on by clifford via conjugation.
+        :return: A PauliTerm corresponding to pauli_in * clifford * pauli_in^{\dagger}
+        """
+        payload = self._clifford_application_payload(clifford, pauli_in)
+        phase_factor, paulis = post_json(self.session, self.sync_endpoint + "/clifford-application", payload).json()
+        pauli_out = PauliTerm("I", 0, 1.j ** phase_factor)
+        for i, pauli in enumerate(paulis):
+            pauli_out *= PauliTerm(pauli, i)
+        return pauli_out * pauli_in.coefficient
+
+    def _rb_sequence_payload(self, depth, qubits, gateset):
+        """
+        Prepares a JSON payload for generating a randomized benchmarking sequence - see generate_rb_sequence.
+
+        :param int depth: The number of cliffords per rb sequences to generate.
+        :param int qubits: The number of qubits to perform rb on.
+        :param list gateset: A list of Gate objects that make up the gateset to decompose the Cliffords into.
+        :return: The JSON payload, with keys "depth", "qubits", and "gateset".
+        """
+        payload = {"depth": depth,
+                   "qubits": qubits,
+                   "gateset": [gate.out() for gate in gateset]}
+        return payload
+
+    def generate_rb_sequence(self, depth, qubits, gateset):
+        """
+        Construct a randomized benchmarking experiment on the given qubits, decomposing into gateset.
+
+        The JSON payload that is parsed is a list of lists of indices, or Nones. In the former case, they are the index
+         of the gate in the gateset.
+        :param depth: The number of Clifford gates to include in the randomized benchmarking experiement.
+        :param qubits: The list of qubit indices to use.
```

Comment:
is this a list or an int? Right now it's inconsistent with `_rb_sequence_payload`

Reply:
int, changed.